src/ToDoView.test.js

```diff
@@ -1,8 +1,76 @@
 import { render, screen } from '@testing-library/react';
 import ToDoView from './ToDoView';
+import userEvent from "@testing-library/user-event";
 
-test('renders learn react link', () => {
-    render(<ToDoView />);
-    const linkElement = screen.getByText(/learn react/i);
-    expect(linkElement).toBeInTheDocument();
-});
+const addButton = () => screen.getByRole('button');
+const dialogBox = () => screen.getByLabelText('Add New To Do');
```

Comment:
This isn't really the dialog, it's the title of the dialog. It's also only used once in your tests, so it's not really worth extracting.